hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MetricsRegionServerWrapperImpl.java

```diff
@@ -152,14 +151,15 @@ public MetricsRegionServerWrapperImpl(final HRegionServer regionServer) {
   }
 
   private void initBlockCache() {
-    this.blockCache = this.regionServer.getBlockCache();
-    this.cacheStats = this.blockCache.map(BlockCache::getStats);
-    if (this.cacheStats.isPresent()) {
-      if (this.cacheStats.get() instanceof CombinedBlockCache.CombinedCacheStats) {
-        l1Stats = Optional
-            .of(((CombinedBlockCache.CombinedCacheStats) this.cacheStats.get()).getLruCacheStats());
-        l2Stats = Optional.of(((CombinedBlockCache.CombinedCacheStats) this.cacheStats.get())
-            .getBucketCacheStats());
+    this.blockCache = this.regionServer.getBlockCache().isPresent() ?
+      this.regionServer.getBlockCache().get() : null;
+    this.cacheStats = this.blockCache != null ? this.blockCache.getStats() : null;
```

Comment:
Ditto.

Reply:
Thanks @Apache9 for the review. For blockCache and mobFileCache, did orElse() as per your suggestion. Here, since we are doing blockCache null check, orElse() might not work